src/chainstate/stacks/miner.rs

```diff
@@ -816,56 +1117,75 @@ impl StacksBlockBuilder {
                         // once we've hit the runtime limit once, allow boot code contract calls, but do not try to eval
                         //   other contract calls
                         if !cc.address.is_boot_code_addr() {
-                            return Err(Error::StacksTransactionSkipped);
+                            return TransactionResult::skipped(
+                                &tx,
+                                "BlockLimitFunction::CONTRACT_LIMIT_HIT".to_string(),
+                                true,
+                            );
                         }
                     }
                     TransactionPayload::SmartContract(_) => {
-                        return Err(Error::StacksTransactionSkipped)
+                        return TransactionResult::skipped(
+                            &tx,
+                            "BlockLimitFunction::CONTRACT_LIMIT_HIT".to_string(),
+                            true,
+                        );
                     }
                     _ => {}
                 }
             }
-            BlockLimitFunction::LIMIT_REACHED => return Err(Error::StacksTransactionSkipped),
+            BlockLimitFunction::LIMIT_REACHED => {
+                return TransactionResult::skipped(
+                    &tx,
+                    "BlockLimitFunction::LIMIT_REACHED".to_string(),
+                    true,
+                )
+            }
             BlockLimitFunction::NO_LIMIT_HIT => {}
         };
 
         let quiet = !cfg!(test);
-        let receipt = if !self.anchored_done {
+        let result = if !self.anchored_done {
             // building up the anchored blocks
             if tx.anchor_mode != TransactionAnchorMode::OnChainOnly
                 && tx.anchor_mode != TransactionAnchorMode::Any
             {
-                return Err(Error::InvalidStacksTransaction(
+                return TransactionResult::skipped(
```

Comment:
is this a change in logic from error to skipped?

Reply:
This isn't exactly a "transaction error", but I agree that the error passed up from this method should be the same. I think this another case where the `::skipped` variant should be constructed with an Error type.